reference/security.rst

```diff
@@ -0,0 +1,291 @@
+the scope (or caveats) of what a macaroon is entitled to.
+
+These operations are performed using ``HMAC-SHA256`` and ``XSalsa20-Poly1305``. The following Go 
+packages are used by the underlying macaroon package for these operations:
+
+- ``crypto/hmac``
+- ``crypto/sha256``
+- `<golang.org/x/crypto/nacl/secretbox>`__
+
+Additionally, the higher-level `Macaroon Bakery package <https://github.com/go-macaroon-bakery/macaroon-bakery>`__
+is used to interface with macaroons and introduces public key cryptography to perform similar operations
+as mentioned above. This allows services to trust macaroons generated externally.
+
+These operations are performed using ``Ed25519`` and ``XSalsa20-Poly1305``. The following Go packages are 
+used by the underlying macaroon bakery package for these operations:
+
+- `<golang.org/x/crypto/nacl/box>`__
+- `<golang.org/x/crypto/curve25519>`__
+
+When a Juju controller is connected to JAAS, the ``login-token-refresh-url`` is used to determine where 
+the JAAS macaroon public key is located. This public key is used when Juju controllers issue macaroons 
+and enforces that the macaroon can only be  discharged by JAAS, who holds the private key. Discharging 
+a macaroon refers to the process of verifying its claims.
+
+Specific details are below:
+
+- **Macaroon Public Key endpoint:** ``<jimm-url>/macaroons/publickey``
+- **Key Type:** Ed25519 (256-bit key)
+- **Signing algorithm:** Ed25519
+
+TLS Communication
+-----------------
+
+In this section we will cover the use of TLS between components of JAAS.
+
+TLS encryption is handled by the Go standard library packages:
+
+- ``crypto/tls``
+- ``crypto/x509``
+
+Client - JAAS
+^^^^^^^^^^^^^
+
+The Juju client package enforces the use of TLS when connecting to a controller.
+This extends to both the Juju CLI and the Juju Terraform Provider.
+
+The minimum supported version is TLS v1.2
+
+JAAS - Juju Controllers
+^^^^^^^^^^^^^^^^^^^^^^^
+
+JAAS enforces the use of TLS when connecting to a Juju controller.
+
+The minimum supported version is TLS v1.2
+
+JAAS - OpenFGA
+^^^^^^^^^^^^^^
+
+JAAS does not currently enforce TLS when communicating with OpenFGA.
+
+TLS is not currently supported with the OpenFGA charm operator.
+
+JAAS - Vault
+^^^^^^^^^^^^
+
+JAAS assumes that Vault is reachable with TLS but does not enforce this.
+By default the Vault charm employs the use of TLS.
+
+The minimum supported version is TLS v1.2.
+
+JAAS - PostgreSQL
+^^^^^^^^^^^^^^^^^
+
+JAAS does not currently enforce TLS when communicating with PostgreSQL.
+But this can be achieved when using the PostgreSQL charm.
+
+The minimum supported version is TLS v1.2.
+
+CORS
+----
+
+CORS or Cross-Origin Resource Sharing is a browser security feature designed to prevent
+malicious use of your online credentials. Read more on CORS 
+`here <https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#what_requests_use_cors>`__.
+
+JAAS supports the use of CORS headers specifically for the purposes of the Juju
```

Comment:
Do we want to add that CORS is not supported for WS communications?

Reply:
Yes I'll need to mention that.